### PR TITLE
#2274 - In-place scale for zonotopes

### LINKED ISSUE #2274: Concrete in-place scaling of a zonotope

### PR DESCRIPTION
**Pull request by mforets**
Closes #2274.

### CODE REVIEW

File: test/unit_Zonotope.jl [hunk 1]
```diff
@@ -90,6 +90,11 @@ for N in [Float64, Rational{Int}, Float32]
     @test Z5.center == N[0, 1]
     @test Z5.generators == N[0.5 0.5 0.5 0; -0.5 0.5 0 0.5]
 
+    # in-place scale
+    Z5aux = copy(Z3)
+    scale!(0.5, Z3)
+    @test Z5 ≈ Z5aux
```
Comment (schillic) [inline test/unit_Zonotope.jl:96]: I do not understand this test. Is there a typo?

File: test/unit_Zonotope.jl [hunk 2]
```diff
@@ -90,6 +90,11 @@ for N in [Float64, Rational{Int}, Float32]
     @test Z5.center == N[0, 1]
     @test Z5.generators == N[0.5 0.5 0.5 0; -0.5 0.5 0 0.5]
 
+    # in-place scale
+    Z5aux = copy(Z3)
+    scale!(0.5, Z3)
```
Comment (mforets) [inline test/unit_Zonotope.jl:95]:
```suggestion
    scale!(0.5, Z5aux)
```

File: test/unit_Zonotope.jl (hunk 1 above)
Comment (mforets) [reply test/unit_Zonotope.jl]: yes, thanks

File: test/unit_Zonotope.jl [hunk 3]
```diff
@@ -90,6 +90,11 @@ for N in [Float64, Rational{Int}, Float32]
     @test Z5.center == N[0, 1]
     @test Z5.generators == N[0.5 0.5 0.5 0; -0.5 0.5 0 0.5]
 
+    # in-place scale
+    Z5aux = copy(Z3)
+    scale!(0.5, Z5aux)
```
Comment (mforets) [inline test/unit_Zonotope.jl:95]:
```suggestion
    scale!(N(0.5), Z5aux)
```

**Comment by mforets** [comment]
the error is a missing method,

> MethodError: no method matching _isapprox(::Array{Float64,2}, ::Array{Float64,2})

i'll open an issue, and try with `isequivalent` in the meantime (though it may also need that method?)

File: test/unit_Zonotope.jl [hunk 4]
```diff
@@ -90,6 +90,11 @@ for N in [Float64, Rational{Int}, Float32]
     @test Z5.center == N[0, 1]
     @test Z5.generators == N[0.5 0.5 0.5 0; -0.5 0.5 0 0.5]
 
+    # in-place scale
+    Z5aux = copy(Z3)
+    scale!(N(0.5), Z5aux)
+    @test Z5 ≈ Z5aux
```
Comment (mforets) [inline test/unit_Zonotope.jl:96]:
```suggestion
    @test isequivalent(Z5, Z5aux)
```

**Comment by mforets** [comment]
> i'll open an issue, and try with isequivalent in the meantime (though it may also need that method?)

yes, that's also required so i think we should better add that method (we hit this today again in https://github.com/JuliaReach/LazySets.jl/pull/1628/files#diff-4f382d7499209ddbd463488f4f6f2ce2R262)

**Comment by mforets** [comment]
> i'll open an issue, and try with isequivalent in the meantime (though it may also need that method?)

this method has now been added so i'll try rebasing..

**Comment by mforets** [comment]
nothing has changed from that last review (aside from adding the missing a missing `export` statement), so this shall be ready to merge after tests pass..

File: src/Sets/Zonotope.jl [hunk 5]
```diff
@@ -313,6 +314,28 @@ function scale(α::Real, Z::Zonotope)
...
+Concrete scaling of a zonotope modifing `Z` in-place
+
+### Input
+
+- `α` -- scalar
+- `Z` -- zonotope
+
+### Output
+
+The zonotope `Z` after applying the numerical scale `α` to is center and generators.
```
Comment (mforets) [inline src/Sets/Zonotope.jl:329]:
```suggestion
The zonotope `Z` after applying the numerical scale `α` to its center and generators.
```

**Pull request merged**